calibr8/core.py

```diff
@@ -619,7 +619,7 @@ def save(self, filepath: os.PathLike):
             model_type=".".join([self.__module__, self.__class__.__qualname__]),
             theta_names=tuple(self.theta_names),
             theta_bounds=tuple(self.theta_bounds),
-            theta_guess=tuple(self.theta_guess),
+            theta_guess=tuple(self.theta_guess) if self.theta_guess is not None else {},
```

Comment:
What is the type annotation for the `theta_guess` attribute? Should be something like `Optional[Tuple[float, ...]]` right?

In any case it shouldn't be set to a `dict`.

```suggestion
            theta_guess=tuple(self.theta_guess) if self.theta_guess is not None else None,
```

Reply:
Since the None Type is not iterable the load function won't work with that suggestion. Setting theta_guess to a dict or list works however.    
How do you feel about this?
```suggestion
            theta_guess=tuple(self.theta_guess) if self.theta_guess is not None else [],
```
I'm open for other recommendations 😸

Reply:
IMO the load function should be modified to tolerate `theta_guess=null` entries when loading the JSON

Reply:
In that case we may change `save()` and `load()` by assigning lists and not tuples to `theta_names`, `theta_bounds`, `theta_guess` in the data dict.

Reply:
If the value in the JSON is `null` it will be parsed as `None` and that's what should be assigned

We're using tuples for these attributes instead of lists, because tuples are immutable, and we don't want people to add/remove bounds or anything like that